packages/rapid-ui/src/components/primitives/link/Link.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { RapidStyles } from '../../../utils';
+type LinkType = 'a';
+interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+	styles?: string;
+	as?: LinkType;
+}
+
+const THEME_CLASSNAME = 'rapid-link';
+
+const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
+	({ styles, as = 'a', ...rest }, ref) => {
+		const Component = (type: LinkType) => {
+			switch (type) {
+				case 'a':
+					return (
+						<a
+							ref={ref}
+							className={RapidStyles(
+								styles || rest.className,
+								'', //defaultStyles
+								THEME_CLASSNAME,
+							)}
+						/>
+					);
+			}
+		};
+		return Component(as);
+	},
+);
+
+Link.displayName = 'Link';
+
+export default Link;
```

Comment:
It's cool we're supporting the use of as here, but instead of using a switch statement I think we can make it a bit more dynamic.

```

interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
  styles?: string;
  as?: React.ElementType; // Allow any component type
}

const THEME_CLASSNAME = 'rapid-link';

const Link = React.forwardRef<any, LinkProps>(
  ({ as: Component = 'a', styles, className, ...rest }, ref) => {
    return (
      <Component
        ref={ref}
        className={RapidStyles(styles || className, '', THEME_CLASSNAME)}
        {...rest}
      />
    );
  }
);

Link.displayName = 'Link';

export default Link;
```

This also spreads the `...rest` prop onto the anchor element to make sure all other props are passed through.